k8s/webpackager/README.md

```diff
@@ -0,0 +1,308 @@
+a new namespace:
+
+```shell
+kubectl create namespace "$NAMESPACE"
+```
+
+#### Expand the manifest template
+
+Use `helm template` to expand the template. We recommend that you save the
+expanded manifest file for future updates to the application.
+
+```shell
+helm template chart/webpackager \
+  --name "$APP_INSTANCE_NAME" \
+  --namespace "$NAMESPACE" \
+  --set replicaCount="$REPLICAS" \
+  --set image.repo="$IMAGE_WEBPACKAGER" \
+  --set image.tag="$TAG" \
+  --set init.repo="$IMAGE_WEBPACKAGER_INIT" \
+  --set init.tag"$TAG" \
+  --set packager.persistence.storageClass="$STORAGE_CLASS" \
+  --set packager.persistence.size="$PERSISTENT_DISK_SIZE" \
+  --set packager.domain="$WEBPACKAGER_DOMAIN" \
+  --set packager.country="$WEBPACKAGER_COUNTRY" \
+  --set packager.locality="$WEBPACKAGER_LOCALITY" \
+  --set packager.organization="$WEBPACKAGER_ORGANIZATION" \
+  --set acme.emailAddress="$ACME_EMAIL_ADDRESS" \
+  --set acme.directoryUrl="$ACME_DIRECTORY_URL" \
+  --set service.loadBalancerSourceRanges="$WEBPACKAGER_LOAD_BALANCER_RANGE" \
+  > "${APP_INSTANCE_NAME}_manifest.yaml"
+```
+
+#### Apply the manifest to your Kubernetes cluster
+
+Use `kubectl` to apply the manifest to your Kubernetes cluster:
+
+```shell
+kubectl apply -f "${APP_INSTANCE_NAME}_manifest.yaml" --namespace "${NAMESPACE}"
+```
+
+#### View the app in the Google Cloud Console
+
+To get the GCP Console URL for your app, run the following command:
+
+```shell
+echo "https://console.cloud.google.com/kubernetes/application/${ZONE}/${CLUSTER}/${NAMESPACE}/${APP_INSTANCE_NAME}"
+```
+
+To view your app, open the URL in your browser.
+
+# Using the app
+
+You can get the IP addresses for your Web Packager solution either from the
+command line, or from the Google Cloud Platform Console.
+
+In the GCP Console, do the following:
+
+1.  Open the
+    [Kubernetes Engine Services](https://console.cloud.google.com/kubernetes/discovery)
+    page.
+1.  Identify the Web Packager solution using its name (typically `webpackager-service`,
+    unless you decide to name your instance some other name).
+1.  From the Endpoints column, note the IP addresses for ports 6000.
+
+If you are using the command line, run the following command:
+
+```shell
+kubectl get svc -l app.kubernetes.io/name=$APP_INSTANCE_NAME --namespace "$NAMESPACE"
+```
+
+This command shows the internal and external IP address of your Web Packager
+service.
+
+You can follow [Web Packager Productionizing][Production] for further steps.
+
+[Production]:https://github.com/google/webpackager/tree/master/cmd/webpkgserver
+
+# Scaling
+
+By default, the Web Packager application is deployed using 2 replicas. You can
+manually scale it up or down using the following command:
+
+```shell
+kubectl scale statefulsets "$APP_INSTANCE_NAME-webpackager" \
+  --namespace "$NAMESPACE" \
+  --replicas=[NEW_REPLICAS]
+```
+
+where `[NEW_REPLICAS]` is the new number of replicas.
+
+# Uninstalling the app
```

Comment:
Any instructions/tips you can provide on updating the app? (e.g. How to do a rolling update that doesn't require an outage; how to update without losing the state in the persistent volume.)

Reply:
Added a comment about not deleting persistent volume if data retention is desired.